content/docs/cml/self-hosted-runners.md

```diff
@@ -83,24 +83,27 @@ you do it your way :)
+| `--repo`                 | Specifies the Git repository to be used. If not specified, repo is inferred from the environmental variables. | Example: `https://github.com/iterative/cml`                                                                    |
+| `--token`                | Personal access token to be used. If not specified, it will be inferred from the environment.                 | `token` should be a string                                                                                     |
+| ` --cloud`               | Cloud provider to deploy the runner                                                                           | Choices: `aws`,`azure`                                                                                         |
+| `--cloud-region`         | Region where the instance is deployed.                                                                        | choices: `us-east`,`us-west`, `eu-west`, `eu-north`. Also accepts native cloud regions. Defaults to `us-west`. |
+| `--cloud-type`           | Instance type.                                                                                                | Choices: `m`, `l`, `xl`. Also supports native types like i.e. `t2.micro`                                       |
+| `--cloud-gpu`            | GPU type.                                                                                                     | Choices: `nogpu`,`k80`,`tesla`                                                                                 |
+| `--cloud-hdd-size`       | HDD size in GB.                                                                                               | Accepts integer values                                                                                         |
+| `--cloud-ssh-private`    | Your private RSA SSH key. If not provided will be generated by the Terraform-provider-Iterative.              | Accepts string                                                                                                 |
+| `--cloud-spot`           | Request a spot instance                                                                                       | Boolean                                                                                                        |
+| `--cloud-spot-price`     | Spot max price. If not specified it takes current spot bidding pricing.                                       | Default: `-1`                                                                                                  |
+| `--cloud-startup-script` | Run the provided script as part of the instance initialization                                                | String (Base64 encoded shell script)                                                                           |
```

Comment:
```suggestion
| `--cloud-startup-script` | Run the script provided in-line during the instance initialization. Expects a Base64-encoded shell script                                                | String                                                                           |
```

Reply:
> the instance initialization

Is it only one instance always? Or should it be more general in case of multiple runners e.g. "during runner instance(s) initialization?

Also, would it be important to specify when exactly during initialization this is run?

Reply:
> shell script

Maybe "Linux shell script"? Or can it be other stuff e.g. BAT (Windows). Also if it's specifically Bash, maybe say "Bash script".

Reply:
> _Maybe "Linux shell script"?_

For the _connoisseur_, the term [_shell script_](https://en.wikipedia.org/wiki/Shell_script) doesn't leave any doubt about its more-or-less POSIX-compliant nature, but this is just one of my attrocious nitpicking exercises. 😂  I would be tempted to use Bash as you suggested, but the fun fact is that we're using the good old [Bourne shell](https://en.wikipedia.org/wiki/Bourne_shell).

_Linux shell script_ might be the most helpful term for now, but I'll consider migrating to Bash under the hood and updating the documentation. The good old `sh` shell is not precisely famous for its syntactic sugar.

Reply:
> _Is it only one instance always? Or should it be more general in case of multiple runners e.g. "during runner instance(s) initialization?_

Currently we only support launching a single instance per invocation, so the singular would be pretty safe.

Reply:
> _Also, would it be important to specify when exactly during initialization this is run?_

The technically inaccurate answer is ["nearly in the middle"](https://github.com/iterative/terraform-provider-iterative/blob/d54e3415ba94fcc7b7c28a9b989b4a21289950dc/iterative/resource_runner.go#L225-L227) between the instance configuration and the runner execution:

1. Install all the base packages and configure the instance operating system if required.
2. Run the custom script and wait until it finishes.
3. Configure and launch the CML self-hosted runner.

Reply:
OK no need for so much detail in docs involving the script execution order I guess. I edited my suggestion above to include your other answers. Thanks!

Reply:
Should we link to https://www.base64decode.org/ or similar tool from "Base64-encoded" ? Or perhaps this is explained/linked to from some other CML doc.

Reply:
As far as I know, the CML documentation doesn't include any user-friendly explanation of Base64. Anyhow, I would prefer not to link to any specific on-line tool.

If you consider it neccessary, we can recommend the use of the `base64` command-line tool and link to something neutral, like the pertaining Wikipedia article, but I presume that users searching for this feature will be somewhat familiar with this encoding.

Reply:
> CML documentation doesn't include any user-friendly explanation of Base64

Might be something to make an issue to improve on. Probably not a good idea to assume users will know what it even means.

Reply:
Agreed! In fact, we probably should start by reviewing if the special character issues we're solving with Base64 are worth the impact from the user experience standpoint.

Note to myself: write a nice issue about this concern on the CML repository.

Reply:
Up to you guys @0x2b3bfa0 @DavidGOrtega unfort. currently the docs team has to focus mostly on DVC so can't follow up on this, but hopefully soon we will be able to expand a little and help you more closely 🙂

Reply:
Awesome! Moved to https://github.com/iterative/terraform-provider-iterative/issues/91 in the meanwhile